Providers/Scripts/2.4x-2.5x/Scripts/nxOMSAuditdPlugin.py

```diff
@@ -99,17 +134,59 @@ def GetOMSAgentPidFilePath(WorkspaceId):
 def GetTmpAuomsConfPath(WorkspaceId):
     return os.path.join(GetTmpDir(WorkspaceId), "auoms.conf")
 
+def GetTmpAuomscollectConfPath(WorkspaceId):
+    return os.path.join(GetTmpDir(WorkspaceId), "auomscollect.conf")
+
 def GetTmpAuomsOutputConfPath(WorkspaceId):
     return os.path.join(GetTmpDir(WorkspaceId), "output.conf")
 
 def GetAuomOutputConfPath(WorkspaceId):
     return os.path.join(AUOMS_OUTCONF_DIR, WorkspaceId + ".conf")
 
 def GetTmpAuditRulesPath(WorkspaceId):
-    return os.path.join(GetTmpDir(WorkspaceId), "auditd_plugin.rules")
+    return os.path.join(GetTmpDir(WorkspaceId), TMP_RULES_FILE)
 
 def GetTmpAuditLoadedRulesPath(WorkspaceId):
-    return os.path.join(GetTmpDir(WorkspaceId), "auditd_loaded.rules")
+    return os.path.join(GetTmpDir(WorkspaceId), TMP_LOADED_RULES_FILE)
+
+def IsAuditdRunning():
+    args = []
+    args.append("/usr/bin/pgrep")
+    args.append("-x")
+    args.append("-U")
+    args.append("0")
+    args.append("auditd")
+    proc = subprocess.Popen(args, -1, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+    exit_code = proc.wait()
+    if exit_code == 0:
+        return True
+    return False
+
+def IsAuomsRunning():
+    args = []
+    args.append("/usr/bin/pgrep")
+    args.append("-x")
+    args.append("-U")
+    args.append("0")
+    args.append("auoms")
+    proc = subprocess.Popen(args, -1, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+    exit_code = proc.wait()
+    if exit_code == 0:
+        return True
+    return False
+
+def IsAuomsCollectRunning():
+    args = []
+    args.append("/usr/bin/pgrep")
+    args.append("-x")
+    args.append("-U")
+    args.append("0")
+    args.append("auomscollect")
+    proc = subprocess.Popen(args, -1, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+    exit_code = proc.wait()
+    if exit_code == 0:
+        return True
+    return False
```

Comment:
Consider defining an utility function for checking the a given process name is running then call it in these three defs